lib/create-contentful-api.ts

```diff
@@ -590,17 +582,17 @@ export default function createContentfulApi<OptionType extends ChainOptions>(
     Fields extends FieldsType,
     Locales extends LocaleCode,
     Options extends ChainOptions
-  > = Options extends ChainOptionWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks
+  > = Options extends ChainOption<'allLocales'>
```

Comment:
Fot this piece of code and all similar type checking with ternaries: I don't know how the control flow works here and I'm wondering if it might be exiting where we don't want it to. I mean a situation like this:

```
const x, y;
if (x === 'foo' | y === 'bar') {
return something
} else if (x === 'foo') {
 return something else
}
return something something
```
In this example it will never enter `else if` because first `if` catches it first. The order of the flow should be changed in this case.

Is this a thing we should be careful with here?

Also since `Options extends ChainOption` is not last, does it mean the flow will never enter `Options extends ChainOption<'noUnresolvableLinks'>`?

Reply:
Fortunately, there is no way to skip ahead like that in the ternaries here. There’s no `return`, `break`, `continue` or similar. So we’re safe there.

> Also since `Options extends ChainOption` is not last, does it mean the flow will never enter `Options extends ChainOption<'noUnresolvableLinks'>`?

`ChainOption` is not a catch-all type, it’s the shorthand for `ChainOption<undefined>`. Thus `Options extends ChainOption` only evaluates to true if all options are `false` (the unconfigured/default client case). The order only matters if some types we check are sub types of each other.